crates/primitives/src/transaction/mod.rs

```diff
@@ -71,6 +71,18 @@ pub(crate) static PARALLEL_SENDER_RECOVERY_THRESHOLD: Lazy<usize> =
         _ => 5,
     });
 
+/// Minimum length of a rlp-encoded legacy transaction.
+pub const MIN_LENGTH_LEGACY_TX_ENCODED: usize = 10;
+/// Minimum length of a rlp-encoded eip2930 transaction.
+pub const MIN_LENGTH_EIP2930_TX_ENCODED: usize = 14;
+/// Minimum length of a rlp-encoded eip1559 transaction.
+pub const MIN_LENGTH_EIP1559_TX_ENCODED: usize = 15;
+/// Minimum length of a rlp-encoded eip4844 transaction.
+pub const MIN_LENGTH_EIP4844_TX_ENCODED: usize = 17;
+/// Minimum length of a rlp-encoded deposit transaction.
+#[cfg(feature = "optimism")]
+pub const MIN_LENGTH_DEPOSIT_TX_ENCODED: usize = 65;
```

Comment:
how did we get these?

Reply:
the tests as mentioned in the description of this pr